src/solver/optim/api/include/antares/solver/optim/api/hasName.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+
+#pragma once
+
+#include <string>
+
+namespace Antares::Solver::Optim::Api
+{
+
+/// Inherited by IMipVariable and IMipConstraint
+class IHasName
+{
+public:
+    virtual ~IHasName() = default;
+    virtual const std::string& getName() const = 0;
```

Comment:
About **getName()** : 
We could rename it into **name()**, shorter and expressive enough.

Reply:
about that, I think we should have a common convention for getters / setters and stick to it in all the code